Silence deprecation warning for collections.Mapping

seibert opened the pull request:
Starting with Python 3.7 (maybe earlier?) a DeprecationWarning is generated:

> /Users/sseibert/anaconda/envs/mg/lib/python3.7/site-packages/donfig/config_obj.py:30: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
>     from collections import Mapping

To preserve Python 2.7 support, this PR tries first to import from the currently preferred module, then falls back to the old module if that fails.

djhoese commented:
Thanks! Perhaps I'll make one more patch release and then officially drop python 2 support. This is based on dask's config module and they haven't support python 2 for a while now.

djhoese commented:
Or...maybe we drop python 2.7 support now since our tests can't easily run with it anymore.

seibert commented:
Given that this PR fixes a warning that only exists in Python 3, there isn't any advantage to doing a patch release for Python 2.7 with this PR included.  If you are happy to drop Python 2.7, I can convert this to a one line PR.

djhoese commented:
Sounds good. Go for python 3 only and you can remove python 2 tests from the travis config and appveyor. If you aren't comfortable/familiar with doing that then that's fine I can update it.

seibert commented:
ok, updated commit (apologies for force-push, but didn't see the need to leave the old approach in commit history), removed Python 2.7 from test configurations of Travis and Appveyor, and also raised Python requirement to >=3.4 in setup.py

djhoese commented on setup.py:
```diff
@@ -44,5 +44,5 @@
       install_requires=['pyyaml'],
       tests_require=['pytest'],
-      python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
+      python_requires='>=3.4,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
```
You can remove everything after the `3.4` here since they are included in the new condition. And actually can you make this `3.6`?

seibert replied:
Done
